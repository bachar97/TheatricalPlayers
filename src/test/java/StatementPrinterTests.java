import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.List;

import static org.approvaltests.Approvals.verify;

public class StatementPrinterTests {

    @Test
    void exampleStatement() {

        HashMap<String, Play> plays = new HashMap<>();
        plays.put("hamlet",  new TragedyInfo("Hamlet"));
        plays.put("as-like",  new ComedyInfo("As You Like It"));
        plays.put("othello",  new TragedyInfo("Othello"));
        Invoice invoice = new Invoice(new Customer("1", "BigCo", 103), List.of(
                new Performance("hamlet", 55),
                new Performance("hamlet", 30),
                new Performance("as-like", 20),
                new Performance("as-like", 35),
                new Performance("othello", 40)));

        var result = invoice.printText(plays);

        verify(result);
    }
}
