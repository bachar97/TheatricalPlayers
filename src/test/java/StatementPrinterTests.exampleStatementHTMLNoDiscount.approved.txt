<!DOCTYPE html>
<html>
<head>
<style>
table, th, td {border: 1px solid black;width: 25%;text-align: center;}
caption{padding-top: 10px;caption-side: bottom;}
</style>
</head>
<body>
<h1>Invoice</h1>
<p><b>Client: </b>BigCo</p>
<table>
<tr>
<th>Piece</th>
<th>Seats Sold</th>
<th>Price</th>
</tr>
<tr>
<td>Hamlet</td>
<td>55</td>
<td>$650.00</td>
</tr>
<tr>
<td>Hamlet</td>
<td>30</td>
<td>$400.00</td>
</tr>
<tr>
<td>As You Like It</td>
<td>35</td>
<td>$580.00</td>
</tr>
<tr>
<td>As You Like It</td>
<td>20</td>
<td>$360.00</td>
</tr>
<tr>
<td>Othello</td>
<td>40</td>
<td>$500.00</td>
</tr>
<tr>
<th colspan="2">Total Owed:</th>
<td>$1,730.00</td>
</tr>
<tr>
<th colspan="2">Fidelity Points Earned:</th>
<td>51</td>
</tr>
<tr>
<th colspan="2">Remaining Fidelity Points:</th>
<td>4</td>
</tr>
<caption><i>Payement is required in under 30 days. Daily newsletters that you can't unsubscribe from will be e-mailed to you if you don't.</i></caption>
</table>
</body>
</html>
